tools/python/mbed_tools/cli/cmsis_mcu_descr.py

```diff
@@ -0,0 +1,163 @@
+#
+# Copyright (c) 2024 Arm Limited and Contributors. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+
+"""
+Subcommands to allow managing the list of CMSIS MCU descriptions that comes with Mbed.
+The MCU description list is used both for generating docs, and for providing information to the code
+about the memory banks present on a device.
+
+MCU descriptions are kept in mbed-os/targets/cmsis_mcu_descriptions.json.  Unlike targets.json5,
+this is a json file as it is updated automatically by code.  MCU descriptions are sourced initially
+from the CMSIS pack index (a resource hosted by ARM), but can also be edited manually after being downloaded.
+This is needed since the index is missing certain MCUs and has wrong information about a few others.
+"""
+from mbed_tools.lib.json_helpers import decode_json_file
+
+import click
+import cmsis_pack_manager
+import humanize
+
+import pathlib
+import os
+import datetime
+import logging
+import json
+from typing import Set, Dict, Any
+
+LOGGER = logging.getLogger(__name__)
+
+# Calculate path to Mbed OS JSON files
+THIS_SCRIPT_DIR = pathlib.Path(os.path.dirname(__file__))
+MBED_OS_DIR = THIS_SCRIPT_DIR.parent.parent.parent.parent
+TARGETS_JSON5_PATH = MBED_OS_DIR / "targets" / "targets.json5"
+CMSIS_MCU_DESCRIPTIONS_JSON_PATH = MBED_OS_DIR / "targets" / "cmsis_mcu_descriptions.json5"
+
+
+# Top-level command
+@click.group(
+    name="cmsis-mcu-descr",
+    help="Manage CMSIS MCU description JSON file"
+)
+def cmsis_mcu_descr():
+
+    # Set up logger defaults
+    LOGGER.setLevel(logging.INFO)
+
+
+def open_cmsis_cache(must_exist=True) -> cmsis_pack_manager.Cache:
+    """
+    Open an accessor to the CMSIS cache.  Also prints how old the cache is.
+    """
+
+    cmsis_cache = cmsis_pack_manager.Cache(False, False)
+
+    index_file_path = pathlib.Path(cmsis_cache.index_path)
+    if not index_file_path.exists() and must_exist:
+        raise RuntimeError("CMSIS device descriptor cache does not exist!  Run 'python -m mbed_tools.cli.main cmsis-mcu-descr reload-cache' to populate it!")
+
+    # Check how old the index file is
+    index_age = humanize.naturaltime(datetime.datetime.fromtimestamp(index_file_path.stat().st_mtime))
+    LOGGER.info("CMSIS MCU description cache was last updated " + index_age)
+
+    return cmsis_cache
+
+
+def get_mcu_names_used_by_targets_json5() -> Set[str]:
+    """
+    Accumulate set of all `device_name` properties used by all targets defined in targets.json5
+    """
+    LOGGER.info("Scanning targets.json5 for used MCU names...")
+    used_mcu_names = set()
+    targets_json5_contents = decode_json_file(TARGETS_JSON5_PATH)
+    for mbed_target, target_details in targets_json5_contents.items():
```

Comment:
Since `mbed_target` isn't used, you can use `.values()` to get only dict values without getting keys.
```suggestion
    for target_details in targets_json5_contents.values():
```